substrate/network-libp2p/src/traits.rs

```diff
@@ -181,36 +136,26 @@ impl NetworkConfiguration {
 		NetworkConfiguration {
 			config_path: None,
 			net_config_path: None,
-			listen_address: None,
+			listen_address: iter::once(AddrComponent::IP4(Ipv4Addr::new(0, 0, 0, 0)))
+				.chain(iter::once(AddrComponent::TCP(30333)))
+				.collect(),
 			public_address: None,
-			udp_port: None,
-			nat_enabled: true,
-			discovery_enabled: true,
 			boot_nodes: Vec::new(),
 			use_secret: None,
 			min_peers: 25,
 			max_peers: 50,
-			max_handshakes: 64,
-			reserved_protocols: HashMap::new(),
-			ip_filter: IpFilter::default(),
 			reserved_nodes: Vec::new(),
 			non_reserved_mode: NonReservedPeerMode::Accept,
 			client_version: "Parity-network".into(),
 		}
 	}
 
-	/// Create new default configuration with specified listen port.
-	pub fn new_with_port(port: u16) -> NetworkConfiguration {
-		let mut config = NetworkConfiguration::new();
-		config.listen_address = Some(SocketAddr::V4(SocketAddrV4::new(Ipv4Addr::new(0, 0, 0, 0), port)));
-		config
-	}
-
-	/// Create new default configuration for localhost-only connection with random port (usefull for testing)
+	/// Create new default configuration for localhost-only connection with random port (useful for testing)
```

Comment:
hm, help me understand, I thought that `AddrComponent::TCP(30333)` should assign the specific port. Why this saying "with random port" then?

Reply:
Oh, I screwed up.